src/chapters/angular-lsn3/attribute-directives.rst

```diff
@@ -0,0 +1,224 @@
+``skill-set.component.css`` files.
+
+.. figure:: ./figures/lesson3-menu.png
+   :alt: Access Angular lesson 3 in VSCode.
+
+Open the terminal panel and navigate to the lesson3 ``examples`` folder. Also
+make sure that you are on the ``master`` branch.
+
+.. sourcecode:: bash
+
+   $ git branch
+      * master
+      solutions
+   $ ls
+      lesson1 lesson2 lesson3
+   $ cd lesson3
+   $ ls
+      examples        exercises
+   $ cd examples
+
+Once you are in the folder, enter ``npm install`` in the terminal, then run
+``ng serve`` to launch the project.
+
+You should see the following:
+
+.. figure:: ./figures/lesson3-attribute-directive-practice-start.png
+   :alt: Attribute directives starting screen.
+
+Update the Skill-Set Styling
+-----------------------------
+
+Examine the code in the three ``skill-set.component`` files:
+
+.. admonition:: Examples
+
+   ``skill-set.component.css``
+
+   .. sourcecode:: css
+      :linenos:
+
+      h3 {
+         text-align: center;
+         text-decoration: underline;
+      }
+
+      .skills {
+         color: green;
+      }
+
+   ``skill-set.component.html``
+
+   .. sourcecode:: html+ng2
+      :linenos:
+
+      <div class="skills">
+         <h3>{{listHeading}}</h3>
+         <ol>
+            <li *ngFor="let skill of skills">{{skill}}</li>
+         </ol>
+         <hr>
+         <h3>Copy of Skill List</h3>
+         <ol>
+            <li *ngFor="let skill of skills">{{skill}}</li>
+         </ol>
+         <hr>
+         <p>Here is some practice text...</p>
+      </div>
+
+   ``skill-set.component.ts`` SkillSetComponent class:
+
+   .. sourcecode:: TypeScript
+      :linenos:
+
+      export class SkillSetComponent implements OnInit {
+         listHeading: string = 'Some Coding Skills I Know';
+         skills: string[] = ['Loops', 'Conditionals', 'Functions', 'Classes', 'Modules', 'Git', 'HTML/CSS'];
+         alternateColor: string = 'black';
+         bulletType: string = 'A';
+         changeColor: boolean = true;
+
+         constructor() { }
+
+         ngOnInit() { }
+
+      }
+
+Right now, there is an awful lot of green on the page, which is set by the
+``skills`` class in the CSS file. Let's fix this with some attribute
+directives.
+
+Inline Styling
+^^^^^^^^^^^^^^^
+
+To change the color and bullet type of the first list element, we could
+override the CSS instructions with some inline code:
+
+.. sourcecode:: html
+
+   <ol style="color: black" type="A">
+
+However, we can use what we learned about data-binding to replace these
+hard-coded styles with variables:
+
+.. sourcecode:: html+ng2
+
+   <ol [style.color]="alternateColor" [type]="bulletType">
+
+Ideas to note:
+
+#. Unlike the structural directives ``*ngFor`` and ``*ngIf``, we can add more
+   than one attribute directive to an HTML tag.
+#. The ``style`` attribute has different properties that can be assigned using
+   dot notation. Examples include ``style.color`` and ``style.background``.
```

Comment:
If hyphens work, is it necessary to draw attention to this detail?  I'm fine either way, but I want to avoid nitty-gritty details that have a low cost/benefit ratio.